Src/VimCore/CoreTypes.fs

```diff
@@ -52,6 +52,27 @@ type JoinKind =
     | RemoveEmptySpaces
     | KeepEmptySpaces
 
+type NavigationKind =
+    | First = 1

```

Comment:
Numbering should start at 0 here to follow the .NET Design Guidelines [https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/enum](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/enum)

Reply:
I initially did not have the numeric assignments and, as a result, the type was a discriminated union instead of an enum. That interoperates poorly with C# and switch, where I ended up with code like this:

```c#
    switch (navigationKind)
    {
        case value when value == NavigationKind.First:
            ...
            break;
    }
```

I eventually figured out than an enum would work just as well in F# and work better in C# (i.e. we can use normal case labels).

Looking at the existing types, there are a couple of F# enums (`Direction`, `WindowKind` and `ModeKind`) that also need to start at zero. I'll fix them at the same time.